src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -869,6 +870,47 @@ public override void EmitMarshallingIL(PInvokeILCodeStreams pInvokeILCodeStreams
         }
     }
 
+    class ComInterfaceMarshaller : Marshaller
+    {
+        protected override void AllocAndTransformManagedToNative(ILCodeStream codeStream)
+        {
+
+            var context = this.Context;
+            ILEmitter emitter = _ilCodeStreams.Emitter;
+
+            // Check if we marshall null pointer
+            ILCodeLabel lNullInterface = emitter.NewCodeLabel();
```

Comment:
It would be better to call a helper method to do this instead of inlining the logic here.

Reply:
Not sure what should I do. If you asking about labels creation, then this is how `BlittableArrayMarshaller` implemented. In a sense for sure, but they are directly created there. I understand that maybe I can create helper method for reporting that scenario not supported (throwing exception code) but you point to different location. I'm lost. Can you clarify?

Reply:
I meant add a method to `InteropHelpers` and just call it from here to do all the work. Many marshalers are implemented like that.

Reply:
Do I properly understand you, that I add code in the `System.Private.CoreLib` and then marshaller will just assume that `classlib` has that method, and just call that method. Am I correctly understand you?

Reply:
Yes, it is exactly what I meant.

Reply:
@jkotas It's magic. Much easier then I initially thought how it should be done!